cli/src/backend/dev/extensions.rs

```diff
@@ -0,0 +1,159 @@
+use std::{fs, path};
+
+use cynic_parser::{type_system as ast, TypeSystemDocument};
+
+#[derive(Debug)]
+pub(super) struct DetectedExtension {
+    pub(super) url: String,
+    pub(super) name: String,
+}
+
+pub(super) fn detect_extensions(parsed_schema: &TypeSystemDocument) -> Vec<DetectedExtension> {
+    let link_directives = parsed_schema
+        .definitions()
+        .filter_map(|definition| match definition {
+            ast::Definition::Schema(schema_definition) | ast::Definition::SchemaExtension(schema_definition) => {
+                Some(schema_definition.directives())
+            }
+            _ => None,
+        })
+        .flatten()
+        .filter(|directive| directive.name() == "link");
+
+    let mut extensions = Vec::new();
+
+    for link_directive in link_directives {
+        let Some(url) = link_directive.argument("url").and_then(|value| value.value().as_str()) else {
+            continue;
+        };
+
+        if url.starts_with("https://specs.apollo.dev") {
+            // for sure not a grafbase extension
+            continue;
+        }
+
+        if let Some(file_path) = url.strip_prefix("file:") {
+            if let Some(extension) = try_load_extension_from_file(file_path, url) {
+                extensions.push(extension);
+            }
+        }
+    }
+
+    tracing::info!(?extensions, "Detected extensions");
+
+    extensions
+}
+
+fn try_load_extension_from_file(file_path: &str, url: &str) -> Option<DetectedExtension> {
```

Comment:
You could just parse the string into a `url::Url` and use `url.to_file_path()`, see https://github.com/grafbase/grafbase/blob/main/crates/extension-catalog/src/load.rs#L13

Reply:
Actually `load_manifest` there is exactly what you need :)

Reply:
Good catch! Should have thought of this.